Assets/Scripts/Managers/DayCycleManager.cs

```diff
@@ -103,22 +113,27 @@ public static void DeregisterEndOfDayCallback( EndOfDayCallback callback )
 
 	void StartMidnightOverlay()
 	{
-		_midnightCoroutine = StartCoroutine( FadeInMidnightOverlay() );
+		StartMidnightOverlay( _blackoutDuration );
+	}
+
+	void StartMidnightOverlay( float overlayTime )
+	{
+		_midnightCoroutine = StartCoroutine( FadeInMidnightOverlay( overlayTime ) );
 	}
 
 	/**
 	 * Used to fade to black at midnight.
 	 */
-	IEnumerator FadeInMidnightOverlay()
+	IEnumerator FadeInMidnightOverlay( float overlayTime )
 	{
 		float elapsedTime = 0.0f;
 		while ( true )
 		{
-			if ( elapsedTime < _blackoutDuration )
+			if ( elapsedTime < overlayTime )
 			{
 				elapsedTime += Time.deltaTime;
 				Color overlayColor = _midnightOverlay.color;
-				overlayColor.a = elapsedTime / _blackoutDuration;
+				overlayColor.a = elapsedTime / overlayTime;
 				_midnightOverlay.color = overlayColor;
```

Comment:
can you replace this with _midnightOverlay.color = midnightOverlay.color.SetAlpha( elapsedTime / overlayTime); ?